api/src/test/java/io/opentelemetry/OpenTelemetryTest.java

```diff
@@ -393,6 +395,12 @@ public LabelSet createLabelSet(
       return null;
     }
 
+    @Nullable
+    @Override
+    public LabelSet createLabelSet(@Nonnull Map<String, String> labels) {
```

Comment:
Consider to remove the other versions for the moment?

Reply:
I'm ok with that if everyone else is. It's definitely a simpler API to have only one method. We can add more over time if people ask for it. The only consideration is the no-op case, where we'd now be forcing people to create an extra Map which will then be thrown away. 
What about having this one, plus a varargs version that requires key-value pairs of arguments, which should cover all possible cases, and let the no-op case not require creation of extra garbage when it's not needed?

Reply:
varargs version creates an extra array as well. Can you show me a short code how you want the `varargs version that requires key-value pairs of arguments`?

Reply:
A single String[] is a very very small amount of memory compared to a Map, and all of the overhead of all the Map.Entry instances (and the buckets, etc).

I had an example of what this would look like in an earlier version of this PR. :) Let me dig it up.


This was the varargs implementation in LabelSetSdk:
```
static LabelSet create(String... keyValuePairs) {
    if (keyValuePairs.length == 0) {
      return EMPTY;
    }
    Preconditions.checkArgument(
        (keyValuePairs.length % 2) == 0,
        "LabelSets must be created with the same number of keys as values.");
    Map<String, String> data = new HashMap<>(keyValuePairs.length / 2);
    for (int i = 0; i < keyValuePairs.length; i++) {
      String key = keyValuePairs[i];
      data.put(key, keyValuePairs[++i]);
    }
    return new AutoValue_SdkLabelSet(Collections.unmodifiableMap(data));
  }
```

Reply:
note: this would be way simpler with java 8 streams.  ;)

Reply:
I think I like that. Does anybody have experience with varargs API? Are there any pros/cons that we should consider?

Reply:
The big thing to watch out for is that you really need to make sure you don't try to mix varargs and lists in overloaded methods. Other than that, it's very easy to use, assuming its well documented. I'd love other opinions, though!

Reply:
as a note, if we have a varargs based creator, we can get rid of the `emptyLabelSet` method on the API, since we get that for free with the empty varargs call.

Reply:
I think you can make the change for the moment until we wait for feedback.

Reply:
ok, I'll put in varargs and we can see how it looks

Reply:
belay that. I'll do varargs as a followup PR